internal/locate/region_request.go

```diff
@@ -551,10 +552,11 @@ func (state *accessFollower) next(bo *retry.Backoffer, selector *replicaSelector
 	}
 
 	reloadRegion := false
+	leaderExhaust := state.IsLeaderExhausted(selector.replicas[state.leaderIdx])
 	for i := 0; i < len(selector.replicas) && !state.option.leaderOnly; i++ {
 		idx := AccessIndex((int(state.lastIdx) + i) % len(selector.replicas))
 		selectReplica := selector.replicas[idx]
-		if state.isCandidate(idx, selectReplica) {
+		if state.isCandidate(idx, selectReplica, leaderExhaust) {
```

Comment:
The `leaderExhaust` which is the result of `state.IsLeaderExhausted` should not be used here in the `isCandidate` function as the  `leaderExhaust` checks `isExhausted` with **2** and the `isCandidate` checks only **1**. The `leaderExhaust` is designed to be only used in the L574 `if selector.targetIdx < 0 {` branch to allow an extra retry on the leader peer if no peers are available.